src/FormModel.php

```diff
@@ -328,11 +331,19 @@ private function generateAttributeLabel(string $name): string
 
     private function readProperty(string $attribute)
     {
-        $class = get_class($this);
+        $class = static::class;
         if (!property_exists($class, $attribute)) {
```

Comment:
Shouldn't you move it below?

Reply:
Yep, missed that one.

Reply:
While looking again on it.... Do we really want to allow getter and setter to work without having attribute at all? `writeProperty` checks for property existence anyway as it is called directly only in `setAttributes` where this check is done.

Reply:
I can surely refactor all that to ignore attribute (property) existence in both cases, and `collectAttributes` can also collect getters and setters in that case. But in that case we won't know whether the intent of getter/setter is to load/get property or some other use case just for model logic.

Reply:
I think we need to use `set{field name}` even if property doesn't exists.
For example property can be named with underscore that reflect private property (old school naming)

Reply:
Yep, but what if I add some setter or getter for internal logic only, see for example `getInflector` method. In this exact case it is private and will be ignored, but I also have public such public method which is only used for mapping to entity for example. In that case we will need another static method to define which attributes (properties) are used in these process and which are not.

Reply:
internal methods should be private

Reply:
I think it will be better to open a separate issue to discuss the usability of allowing getter/setter without actual attribute considering the potential problem I have mentioned above. Also it would be better to address that as a separate PR as it would need a rewrite not only to `readProperty`/`writeProperty` methods, but also `collectAttributes` to collect virtual (getter and setter only without actual attribute) attributes by reading method signature (parameter type of setter or return type of getter). I see that can be helpful sometimes not to have backing attribute at all and have virtual attribute which is just composed attribute for example, e.g. `$this->firstName . ' ' . $this->lastName`. But in that case why do we need `firstName` and `lastName` properties on model at all? I will address all issues that come to my mind in that separate issue.

Reply:
> internal methods should be private

By internal I mean internal to application logic not the class itself. Some `getParam` which is used only in a service that is doing mapping form to entity. The method will be public in that case, but it won't have any meaning in the context of setting/getting parameters.

Reply:
By allowing getter/setter without actual property we are just restricting the naming of methods. We are disallowing the end user to name any method `getSomething` or `setSomething` if the method is not intended to be used in attribute collection process, so we need to weigh out on that.

Reply:
> I think it will be better to open a separate issue

Yes, you are right. #39